quarkus/config-api/src/main/java/org/keycloak/config/database/Database.java

```diff
@@ -123,7 +123,7 @@ public enum Vendor {
                 Enabled.LEGACY_ONLY, 
                 "org.h2.jdbcx.JdbcDataSource",
                 "org.h2.Driver",
-                "io.quarkus.hibernate.orm.runtime.dialect.QuarkusH2Dialect",
+                "org.hibernate.dialect.H2Dialect",
```

Comment:
For more information see: https://github.com/quarkusio/quarkus/wiki/Migration-Guide-3.0:-Hibernate-ORM-5-to-6-migration#dialect-configuration-changes